vignettes/hubEnsembles.Rmd

```diff
@@ -18,167 +18,325 @@ knitr::opts_chunk$set(
+    use_median_as_point = TRUE,
+    interactive = FALSE,
+    intervals = c(0.5, 0.9),
+    pal_color = "Set1",
+    show_legend = TRUE,
+    x_col_name = "target_end_date",
+    x_target_col_name = "date"
+  ) +
+  theme_bw() +
+  labs(y = "incident hospitalizations")
+
+# Pull legend
+l <- cowplot::get_plot_component(p1, "guide-box", return_all = TRUE)[[3]]
+
+# Assemble plots
+cowplot::plot_grid(
+  cowplot::plot_grid(
+    p1 + labs(subtitle = "example PMF output type") +
+      theme(legend.position = "none"),
+    p2 + labs(subtitle = "example quantile output type") +
+      theme(legend.position = "none"),
+    labels = LETTERS[1:2]
+  ), l,
+  ncol = 1,
+  rel_heights = c(0.95, 0.05)
+)
 ```
 
+Like with `simple_ensemble()`, we can change the default aggregation settings. For example, weights that determine a model's contribution to the resulting ensemble may be provided.
+
 ```{r}
-plot_df <- dplyr::bind_rows(model_outputs, mean_ens) %>%
-  dplyr::filter(location == "US", origin_date == "2022-12-12") %>%
-  dplyr::mutate(target_date = origin_date + horizon)
-
-plot <- basic_plot_function(
-  plot_df,
-  truth_df = target_data %>%
-    dplyr::filter(
-      location == "US",
-      time_idx >= "2022-10-01",
-      time_idx <= "2023-03-01"
-    ),
-  forecast_date = "2022-12-12"
+options(width = 300)
+model_weights <- data.frame(
+  model_id = c("MOBS-GLEAM_FLUH", "PSI-DICE", "simple_hub-baseline"),
+  weight = c(0.4, 0.4, 0.2)
 )
-plot
+weighted_linear_pool_norm <- hubExamples::forecast_outputs |>
+  dplyr::filter(!output_type %in% c("median", "sample")) |>
+  hubEnsembles::linear_pool(
+    weights = model_weights,
+    model_id = "linear-pool-weighted"
+  )
+head(weighted_linear_pool_norm, 10)
+```
+
+We can also change the distribution that `distfromq` uses to estimate the component models' distributional tails to either log normal or cauchy using the `tail_dist` argument. See documentation in the `distfromq` package for more details and function options.
+
+```{r}
+linear_pool_lnorm <- hubExamples::forecast_outputs |>
+  dplyr::filter(!output_type %in% c("median", "sample")) |>
+  hubEnsembles::linear_pool(
+    model_id = "linear-pool-lognormal",
+    tail_dist = "lnorm"
+  )
+linear_pool_cauchy <- hubExamples::forecast_outputs |>
+  dplyr::filter(!output_type %in% c("median", "sample")) |>
+  hubEnsembles::linear_pool(
+    model_id = "linear-pool-cauchy",
+    tail_dist = "cauchy"
+  )
+```
+
+The figure below compares three linear pools with different distribution tails — normal, lognormal, and cauchy — predicting weekly incident influence hospitalizations in Massachusetts. The plotted quantiles show that the three linear pools generate similar predictive distributions, except for the model with cauchy tails near the outer edges of the distribution. This is expected given that the only difference between the linear pools is the selection of tail distribution, and cauchy distributions have heavy tails.
```

Comment:
I would vote to remove this text/figure. I don't find it particularly helpful because the resulting ensembles are so similar. This makes the colors difficult to distinguish. Perhaps instead we could add one sentence at the end of the last paragraph that says we don't expect this choice to change the resulting ensemble much (akin to the last sentence of this paragraph).

Reply:
How about "Generally, the choice of tail distribution does not have a large impact on the resulting predictive distribution except in the outer edges of the distribution." for the sentence at the end of the previous paragraph?